docs/core-concepts/atomic-swap-htlc.md

```diff
@@ -0,0 +1,133 @@
+
+1. Monitor the ledgers by processing blocks supplied by blockchain nodes.
+1. Define the transaction details and hand them over to an application on top to send the actual signed transaction into the respective blockchain network.
+
+<iframe 
+    src="https://docs.google.com/presentation/d/e/2PACX-1vTSToxZxKhOjfUWHVL5sjjjyGTdEXubjM3TpOpK-qR5Cjs7b6Tda9ZoX6n_NdM9iqgXBGOtPcPnjHAA/embed?start=false&loop=false&delayms=1000"
+    frameborder="0"
+    width="801"
+    height="480"
+    allowfullscreen="true"
+    mozallowfullscreen="true"
+    webkitallowfullscreen="true">
+</iframe>
+
+## Atomic Swap
+
+Atomic Swaps define a peer to peer exchange of two digital assets, where both parties are in control of their private keys throughout the whole execution of the swap.
+
+The atomicity of the swap is achieved through using [HTLC](#htlc)s as a programmable escrow.
+Both parties require to have the same knowledge of all parameters for the HTLCs on both ledgers before they can start.
+Both parties then have the same functions available within the [HTLC](#htlc)s and can decide to move forward or wait to take their money back. 
+No party has access to both assets at the same time within the boundaries of the atomic swap. 
+
+Note that the party in the cryptographic role of Alice has to redeem first, because she has the actual secret.
+
+## HTLC
+
+HTLC stands for Hash Time Locked Contract. An HTLC is a script or smart contract that locks a digital asset over time.
+HTLCs require a ledger that provides a time lock.
+
+<iframe 
+    src="https://docs.google.com/presentation/d/e/2PACX-1vRYb97VhvfyMa3oFC8CiVagBrNmOioSJpasERtSCi6RMwf0MwxCb1yTJBUm8_ZT9OOw6r-_fBsV23GX/embed?start=false&loop=false&delayms=3000"
+    frameborder="0"
+    width="801"
+    height="480"
+    allowfullscreen="true"
+    mozallowfullscreen="true"
+    webkitallowfullscreen="true">
+</iframe>
+
+HTLCs typically have three steps:
+
+1. **Fund**: Locking up the asset in the HTLC.
+1. **Redeem**: Until the expiry time-stamp is reached the other party has the change to redeem the locke-up assets with the secret.
+1. **Refund**: After the expiry condition has been met, the party that locked up the asset can take it back.
+
+In order to create HTLCs for the execution of an atomic swap, the two parties have to exchange certain parameters:
+
+1. Ledgers, assets and amounts to be swapped.
+1. Redeem and refund identity (address) of the respective parties.
+1. Expiry times to be used for the HTLCs.
+1. Hash of the secret (to be sent from the party in the cryptographic role of Alice to the party in the cryptographic role of Bob).
+
+### HTLCs on different Ledgers
+
+On Bitcoin this can be achieved by using a [Bitcoin script](https://en.bitcoin.it/wiki/Script).
+The Bitcoin HTLC Script is defined in the [COMIT Bitcoin HTLC specification](https://github.com/comit-network/RFCs/blob/master/RFC-005-SWAP-Basic-Bitcoin.adoc): 
+
+```
+OP_IF
```

Comment:
Can we do this inlining automatically with docusaurus?

Reply:
you mean inlining the code from github? 

If this it what you mean: I played around a bit using gists and stuff, but could not really make it work.